packages/next/src/server/lib/find-page-file.ts

```diff
@@ -112,26 +115,36 @@ export function createValidFileMatcher(
    * Match the file if it's a metadata route file, static: if the file is a static metadata file.
    * It needs to be a file which doesn't match the custom metadata routes e.g. `app/robots.txt/route.js`
    */
-  function isMetadataFile(filePath: string) {
+  function isMetadataRouteFile(filePath: string) {
     const appDirRelativePath = appDirPath
       ? filePath.replace(appDirPath, '')
       : filePath
 
-    return isMetadataRouteFile(appDirRelativePath, pageExtensions, true)
+    return _isMetadataRouteFile(appDirRelativePath, pageExtensions, true)
```

Comment:
this function has a different name is also because we want to avoid same name but different functionality which will lead to confusing. Having `_isMetadataRouteFile` and `isMetadataRouteFile` will lead to confusing